### PR TITLE
feat!: New Component: IconList

### LINKED ISSUE #1234: [feat] New Component: Icon list
# [Icon list](https://designsystem.digital.gov/components/icon-list)

## USWDS Version

[2.11.0](https://github.com/uswds/uswds/releases/tag/v2.11.0)

## Description

An icon list reinforces the meaning and visibility of individual list items with a leading icon.

## Requirements (proposed)

**Prop Interface**

```JSX
interface IconListProps = {
  className?: string
  children: React.ReactElement<IconListItemProps>[]
} & JSX.IntrinsicElements['ul']
```

```JSX
interface IconListItemProps = {
  className?: string
  children: React.ReactNode
} & JSX.IntrinsicElements['li']
```

```JSX
interface IconListContentProps = {
  className?: string
  children: React.ReactNode
} & JSX.IntrinsicElements['div']
```

```JSX
interface IconListHeaderProps = {
  type: string // if we want to use the pattern est in ProcessListHeading
  className?: string
  children?: React.ReactNode
} &  JSX.IntrinsicElements[TBD]
```

## Example Implementation (proposed)

```JSX
return (
  <IconList>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">Donate cash when possible.</IconListHeader>
        <p>
          Financial contributions to recognized disaster relief organizations...
        </p>
      </IconListContent>
    </IconListItem>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">
          Confirm what donations are needed.
        </IconListHeader>
        <p>
          Unneeded and unsolicited goods burden local organizations’ ability...
        </p>
      </IconListContent>
    </IconListItem>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">
          Talk to trusted organizations about volunteering.
        </IconListHeader>
        <p>
          Financial contributions to recognized disaster relief organizations...
        </p>
      </IconListContent>
    </IconListItem>
  </IconList>
)
```

**More Details**

- `<ul>`root element
- Three variants to take into consideration
-   usa-icon-list--[color] - any one of the theme colors listed on the color utilities page
-   usa-icon-list--size-[size] - theme font size: 2xs, xs, sm, md, lg, xl, or 2xl
-   [responsive_variant]:usa-icon-list--size-[size] - Add a responsive breakpoint prefix separated with a : to target a utility at a responsive breakpoint and higher, following a mobile-first methodology.

## Other Considerations

- The responsive variant may be something to put in another story as an enhancement or it might be super easy, I leave that research to the implementor

### PR DESCRIPTION
**Pull request by brandonlenz**
# Summary

Add new component: [IconList](https://designsystem.digital.gov/components/icon-list/)

## Related Issues or PRs

closes #1234
~~fixes a typo in one exported Icon in de855d8d4335645eef6a12a108ac7483ca7438df~~ Since the Icon API was changed to ensure no conflict between `IconList` and the `Icon`, `List`, (previous `IconList`, now `Icon.List`), this is a BREAKING CHANGE.
- When merging and squashing, add the following text:
  > BREAKING CHANGE: Icons no longer follow the pattern `Icon{ICON_NAME}`. They have are now prefixed components using "dot" notation as in `Icon.{ICON_NAME}`. The import therefore also changes, as you now only need to `import { Icon } ...`.
- Note that the side effects of this change are less efficient tree shaking. A consumer using any icon will effectively be importing ALL of the icons.

## How To Test

- Start up storybook (or use the ephemerally deployed storybook instance) and view/inspect the IconList examples.
- Verify Unit test coverage is acceptable and passing

### Screenshots (optional)

image

image

image

image

image

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -65,8 +65,13 @@ export { NavDropDownButton } from './components/header/NavDropDownButton/NavDrop
 export { PrimaryNav } from './components/header/PrimaryNav/PrimaryNav'
 export { Title } from './components/header/Title/Title'
 
+// IconList
+export { IconList as IconListCONFLICT } from './components/IconList/IconList'
```
Comment (haworku) [inline src/index.ts:69]: ^ check this out :)

Comment (brandonlenz) [inline src/index.ts:69]: Leftover relic that I need to fix

Comment (brandonlenz) [reply src/index.ts]: fixed in 2893554. Thanks @haworku who found this in live code review

**Review by haworku — APPROVED** [review]
Approving this. We discussed IRL paying attention to how the change to exporting `{Icon}` influences our bundle size.  This is our best attempt to handle icons for now.

**Comment by brandonlenz** [comment]
Looks like there's some broken behavior of the input prefix/suffix with icons and this story: 
image

I need to check that and fix before merging.

**Comment by brandonlenz** [comment]
> Looks like there's some broken behavior of the input prefix/suffix with icons and this story:
> I need to check that and fix before merging.

Fixed in 9bc9119

**Review by haworku — APPROVED** [review]

**Pull request merged**